### ISSUE TITLE
[BUG] ASoC: cpu DAI prepare error on GLK

### ISSUE THREAD

**Issue by keqiaozhang** [thesofproject/linux#828]
**Describe the bug**
ASoC: cpu DAI prepare error when loading tplg.

**To Reproduce**
This issue happens at boot stage.

**Expected behavior**
Tplg load successful.

**Impact**
backend prepare failed -22

**Environment**
1) Kernel: origin/topic/sof-dev-46c2b852
    Firmware: origin/cml-003-drop-stable-4731a1
2) tplg name：sof-glk-da7219.tplg
3) GLK Chromebook.
4) Reproducibility Rate：100%

**Screenshots or console output**
```
aplay -Dhw:0,5 -f dat -c 2 test.wav
Playing WAVE 'test.wav' : Signed 16 bit Little Endian, Rate 48000 Hz, Stereo
aplay: set_params:1403: Unable to install hw params:
ACCESS:  RW_INTERLEAVED
FORMAT:  S16_LE
SUBFORMAT:  STD
SAMPLE_BITS: 16
FRAME_BITS: 32
CHANNELS: 2
RATE: 48000
PERIOD_TIME: 85000
PERIOD_SIZE: 4080
PERIOD_BYTES: 16320
PERIODS: (4 5)
BUFFER_TIME: 341000
BUFFER_SIZE: 16368
BUFFER_BYTES: 65472
TICK_TIME: 0
```
 **dmesg** 
```[  472.091877] sof-audio-pci 0000:00:0e.0: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[  472.091879] sof-audio-pci 0000:00:0e.0: pcm: stream dir 0, posn mailbox offset is 790528
[  472.091896] sof-audio-pci 0000:00:0e.0: ASoC: cpu DAI prepare error: -22
[  472.091900]  iDisp1: ASoC: backend prepare failed -22
[  472.092197] sof-audio-pci 0000:00:0e.0: pcm: free stream 5 dir 0
```

[dmesg.log](https://github.com/thesofproject/sof/files/3072344/dmesg.log)
[trace.log](https://github.com/thesofproject/sof/files/3072345/trace.log)

**Comment by xiulipan** [comment]
@ranj063 
I had a quick look and the -22 should be returned form following code, right?
https://github.com/thesofproject/linux/blob/46c2b852b448d5bd6b054b60da0e59c5b7720d05/sound/soc/sof/intel/hda-dai.c#L168-L179

> > @ranj063
> > And any idea if we will need to find BE from `snd_sof_find_spcm_dai`?
> 
> @xiulipan no, if you look at sof_load_dai(), we dont create spcm for a BE.
> 
> _Originally posted by @ranj063 in https://github.com/thesofproject/linux/pull/816#issuecomment-482141439_

If `snd_sof_find_spcm_dai` is not designed for find BE, why we are using it here?

**Comment by keqiaozhang** [comment]
Add more logs 
```
[  102.777150] sof-audio-pci 0000:00:0e.0: ipc tx succeeded: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
[  102.777153] sof-audio-pci 0000:00:0e.0: pcm: stream dir 0, posn mailbox offset is 790528
[  102.777170] sof-audio-pci 0000:00:0e.0: PXL: pcm id 7 | rtd id 3
[  102.777172] sof-audio-pci 0000:00:0e.0: PXL: pcm id 6 | rtd id 3
[  102.777174] sof-audio-pci 0000:00:0e.0: PXL: pcm id 5 | rtd id 3
[  102.777176] sof-audio-pci 0000:00:0e.0: PXL: pcm id 99 | rtd id 3
[  102.777177] sof-audio-pci 0000:00:0e.0: PXL: pcm id 1 | rtd id 3
[  102.777179] sof-audio-pci 0000:00:0e.0: PXL: pcm id 0 | rtd id 3
[  102.777181] sof-audio-pci 0000:00:0e.0: ASoC: cpu DAI prepare error: -22
[  102.777183]  iDisp1: ASoC: backend prepare failed -22
```

```diff
diff --git a/sound/soc/sof/sof-priv.h b/sound/soc/sof/sof-priv.h
index 35e78ffecce2..f3c7122b7474 100644
--- a/sound/soc/sof/sof-priv.h
+++ b/sound/soc/sof/sof-priv.h
@@ -481,6 +481,7 @@ struct snd_sof_pcm *snd_sof_find_spcm_dai(struct snd_sof_dev *sdev,
        struct snd_sof_pcm *spcm = NULL;

        list_for_each_entry(spcm, &sdev->pcm_list, list) {
+               dev_err(sdev->dev, "PXL: pcm id %zu | rtd id %zu\n",spcm->pcm.dai_id, rtd->dai_link->id);
                if (le32_to_cpu(spcm->pcm.dai_id) == rtd->dai_link->id)
                        return spcm;
        }
```

**Comment by ranj063** [comment]
> If `snd_sof_find_spcm_dai` is not designed for find BE, why we are using it here?

@xiulipan ack. working on it. will get back asap.

**Comment by libinyang** [comment]
it seems the regression of https://github.com/thesofproject/linux/pull/795?

**Comment by mengdonglin** [comment]
Can be fixed by PR #834, waiting for fixing CI error and getting merged.
